lib/core/text/abstract_text.nit

```diff
@@ -1489,16 +1485,32 @@ abstract class Buffer
 	#     b = new FlatBuffer.from("ab_c -ab0c ab\nc")
 	#     b.capitalize
 	#     assert b == "Ab_C -Ab0C Ab\nC"
-	fun capitalize do
+	#
+	#     b = new FlatBuffer.from("12345")
+	#     b.capitalize(source="foo")
+	#     assert b == "Foo45"
+	#
+	#     b = new FlatBuffer.from("preserve my ACRONYMS")
+	#     b.capitalize(keep_upper=true)
+	#     assert b == "Preserve My ACRONYMS"
+	fun capitalize(source: nullable Text, keep_upper: nullable Bool) do
```

Comment:
The modular `source` is a bit strange I have to agree.

I like the idea of re-using the `Buffer` version to avoid duplicating services, though the implementation is a bit strange, I think it should always work on `self`.
`String::capitalized` can do exactly what is done within this method by pre-appending its contents to the buffer, which might even be faster due to the bulk copy of the original string.